Gordon360/Services/HousingService.cs

```diff
@@ -265,53 +263,60 @@ public int EditApplication(string editorId, string sess_cde, int apartAppId, str
             }
 
             // List of applicants IDs that are in the array recieved from the frontend but not yet in the database
-            List<string> applicantIDsToAdd = new List<string>(newApplicantIDs);
+            List<ApartmentApplicantViewModel> applicantsToAdd = new List<ApartmentApplicantViewModel>(newApartmentApplicants);
 
             // List of applicants IDs that are in both the array recieved from the frontend and the database
-            List<string> applicantIDsToUpdate = new List<string>();
+            List<ApartmentApplicantViewModel> applicantsToUpdate = new List<ApartmentApplicantViewModel>();
 
             // List of applicants IDs that are in the database but not in the array recieved from the frontend
-            List<string> applicantIDsToRemove = new List<string>();
+            List<ApartmentApplicantViewModel> applicantsToRemove = new List<ApartmentApplicantViewModel>();
 
             // Check whether any applicants were found matching the given application ID number
-            if (applicantIDsResult.Any())
+            if (existingApplicantResult.Any())
             {
-                foreach(ApartmentApplicantIDViewModel applicantIDModel in applicantIDsResult)
+                foreach (GET_AA_APPLICANTS_BY_APPID_Result existingApplicant in existingApplicantResult)
                 {
-                    string existingApplicantID = applicantIDModel.ID_NUM;
-                    if (newApplicantIDs.Contains(existingApplicantID))
+                    ApartmentApplicantViewModel newMatchingApplicant = null;
+                    newMatchingApplicant = newApartmentApplicants.FirstOrDefault(x => x.ID == existingApplicant.ID_NUM);  //.Where(x => x.ID == existingApplicantID).First();
+                    if (newMatchingApplicant != null)
                     {
-                        applicantIDsToAdd.Remove(existingApplicantID);
-                        applicantIDsToUpdate.Add(existingApplicantID);
+                        // If the applicant is in both the new applicant list and the existing applicant list, then we do NOT need to add it to the database
+                        applicantsToAdd.Remove(newMatchingApplicant);
+                        if (newMatchingApplicant.OffCampusProgram != existingApplicant.AprtProgram)
+                        {
+                            // If the applicant is in both the new and existing applicant lists but has different OffCampusProgram values, then we need to update that in the database
+                            applicantsToUpdate.Add(newMatchingApplicant);
+                        }
                     }
                     else
                     {
-                        applicantIDsToRemove.Add(existingApplicantID);
+                        // If the applicant is in the existing list but not in the new list of applicants, then we need to remove it from the database
+                        applicantsToRemove.Add(newMatchingApplicant);
```

Comment:
How do we know that newMatchingApplicant being null does not mean it is in the new list but not the existing? Doesn't the FirstOrDefault code give a result only if the value exists in both lists? The "null" could mean something is in the new that is not in the old as well as something is in the old that is not in the new, right?

Reply:
Yes, that is the whole point. newMatching means it is in both, that is the whole point of the check. 

Also, if it is in the new list but not in the existing list, then it will never be seen by the `FirstOrDefault` because that is inside the loop of items from the "existing" list. So if it is not in existing, then it is never a part of this whole foreach loop. It will just stay put in the new list where it already is.

Reply:
It is basically just "for each old item, check if it is also in the new list." 
so 
- the "**add**" list is filled with the "**new**" stuff _**before**_ the loop,
- then any items in both are moved _from_ "**add**" _to_ "**update**",
- any items not in "**new**" (therefore also not in "**add**" or "**update**") will be added to the "**remove**" list

So at the end of all that, all the items are sorted to where they are supposed to be

Reply:
Okay, on a second read-through, this makes total sense.